File: Programowanie obiektowe/Lista5/Zad2/Pomnoz.java
package com.company;

/**
 * Created by teez on 25.03.2016.
 */
public class Pomnoz extends wyrazenie {

    int a;
    int b;


    public Pomnoz(wyrazenie a, wyrazenie b){
        this.a = a.oblicz();
        this.b = b.oblicz();
    }
    public String toString(){

        return " Wynikiem bedzie suma " + this.a + " * " +  this.b;
    }

    public int oblicz() {
        return this.a*this.b;
    }
}

File: Kurs C++/Zad6/pomnoz.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "pomnoz.h"


double pomnoz::oblicz() {
    return a->oblicz()*b->oblicz();
}

pomnoz::pomnoz(wyrazenie *a, wyrazenie *b) {
    this->a = a;
    this->b = b;
}

std::string pomnoz::formatuj_wyrazenie() {
    return "(" + a->formatuj_wyrazenie() + " * " + b->formatuj_wyrazenie() + ")";
}